server/src/socketHandlers.ts

```diff
@@ -45,21 +45,22 @@ export default function initializeSocketHandler(io: Server) {
   };
 
   socket.on("create_lobby", (userId, callback) => {
-   const rooms = Array.from(io.sockets.adapter.rooms).map(
-    ([roomName, sockets]) => roomName
-   );
+   
+   // Returns an array of rooms 
+   const rooms = Array.from(io.sockets.adapter.rooms.keys());
    let room = generateUniqueRoomId(rooms);
 
    lobbies[room] = [userId];
 
    //Returns all the rooms a socket is currently in.
    let socketRoomsArray = Array.from(socket.rooms);
 
-   //Makes sures a socket can only ever be in one room
-   if (socket.rooms.size === 2) {
-    socket.leave(socketRoomsArray[1]);
-    socket.join(room);
-   }
+   //Makes sures a socket can only ever be in one room excluding its own unique room
+   if (socketRoomsArray.length > 1) {
+    for (let i = 1; i < socketRoomsArray.length; i++) {
+      socket.leave(socketRoomsArray[i]);
+        }
+    }
```

Comment:
As you edited the server code, did you try running the tests again? I'm curious if the tests still pass or not. Not because your changes are wrong, but maybe my tests could be too tied to the implementation rather than the behaviour. It should still pass for the create_lobby event, and I hope that it still passes for the join_lobby event. I think socket.io is a little strict about the data that gets passed with the callback.

Reply:
It passed every test except the leave room test.